src/logging.lua

```diff
@@ -110,15 +95,40 @@ end
 -- Creates a new logger object
 -- @param append Function used by the logger to append a message with a
 -- log-level to the log stream.
+-- @param levels optional table of custom logging levels
 -- @return Table representing the new logger object.
 -------------------------------------------------------------------------------
-function logging.new(append)
+function logging.new(append,levels)
   if type(append) ~= "function" then
     return nil, "Appender must be a function."
   end
 
   local logger = {}
   logger.append = append
+	
+	local LEVEL
+	
+	if levels and type(levels) == "table" then
+		LEVEL = levels
+	else
+		LEVEL = DEFAULT_LEVELS
+	end
```

Comment:
I think we should not be using the provided table, but create a copy. While copying, we should convert all entries to uppercase.

and at the end of the `new` function we should verify that non of the names (in either uppercase or lowercase format) collides with an existing field on the new `logger` object. For example if we provide a level "LOG", or "APPEND" then the injected functions will become `logger:log()`, or `logger:append()`, which will override the existing properties/functions.